### PR TITLE
Migrate most of KeyProcessor to SDK types

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
There's still some stuff with old apiclient types because they are used in other files and are a PITA to migrate right now.

**Changes**
- Migrate most of KeyProcessor to SDK types

**Issues**

### CODE REVIEW

**Comment by Andy2244** [comment]
Um can you also mark the old api client types as deprecated?

I'm going mad trying to figure out what import is actually valid and which was just auto added.
The issue is, that often its just upper vs lowercase and hard to spot.

**Comment by nielsvanvelzen** [comment]
The legacy apiclient is unmaintained for over a year now and it's not possible to create new versions for it anymore. So no, I can't add those markers.

File: app/src/main/java/org/jellyfin/androidtv/util/KeyProcessor.java [hunk 1]
```diff
@@ -291,17 +293,17 @@ public static PopupMenu createItemMenu(BaseRowItem rowItem, UserItemDataDto user
 
         //Not sure I like this but I either duplicate processing with in-line events or do this and
         // use a single event handler
-        mCurrentItem = item;
+        mCurrentItem = rowItem.getBaseItem();
         mCurrentRowItemNdx = rowItem.getIndex();
-        mCurrentItemId = item.getId();
+        mCurrentItemId = item.getId().toString();
```
Comment (crobibero) [inline app/src/main/java/org/jellyfin/androidtv/util/KeyProcessor.java:298]: Why remove item from L294 but keep it here?

**Review by crobibero — APPROVED** [review]

File: app/src/main/java/org/jellyfin/androidtv/util/KeyProcessor.java (hunk 1 above)
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/util/KeyProcessor.java:298]: `rowItem.getBaseItem()` is the old BaseItemDto (from the apiclient) which we're migrating from. So I use the SDK one when I can, in this case I can't use it for mCurrentItem but can for the id.

**Pull request merged**